server/webhook.go

```diff
@@ -139,6 +146,193 @@ func (s *Server) registerWebhookRoutes(g *echo.Group) {
 		}
 		return c.String(http.StatusOK, strings.Join(createdMessages, "\n"))
 	})
+
+	// id is the webhookEndpointID in repository
+	// This endpoint is generated and injected into GitHub action & GitLab CI during the VCS setup.
+	g.POST("/sql-review/:id", func(c echo.Context) error {
+		body, err := io.ReadAll(c.Request().Body)
+		if err != nil {
+			return echo.NewHTTPError(http.StatusBadRequest, "Failed to read SQL review request").SetInternal(err)
+		}
+		log.Debug("SQL review request received for VCS project",
+			zap.String("webhook_endpoint_id", c.Param("id")),
+			zap.String("request", string(body)),
+		)
+
+		var request vcsSQLReviewRequest
+		if err := json.Unmarshal(body, &request); err != nil {
+			return echo.NewHTTPError(http.StatusBadRequest, "Malformed SQL review request").SetInternal(err)
+		}
+
+		filter := func(repo *api.Repository) (bool, error) {
+			return c.Request().Header.Get("X-SQL-Review-Token") == repo.WebhookSecretToken && strings.HasPrefix(repo.WebURL, request.WebURL), nil
+		}
+		ctx := c.Request().Context()
+		repositoryList, err := s.filterRepository(ctx, c.Param("id"), request.RepositoryID, filter)
+		if err != nil {
+			return err
+		}
+		if len(repositoryList) == 0 {
+			log.Debug("Empty handle repo list. Ignore this request.")
+			return c.JSON(http.StatusOK, &vcsSQLReviewResult{
+				Status:  advisor.Success,
+				Content: []string{},
+			})
+		}
+
```

Comment:
What shall we do if len(repositoryList) > 1? Maybe return an xml error to notify users that they have an ambiguous setup.

Reply:
I’m not so sure about this. For example, maybe in Project A, users create the VCS repo "Example" with file template `folder1/{{DB_NAME}}_xxxx.sql`, but in Project B, users also create another VCS repo "Example" with file template `folder2/{{DB_NAME}}_xxxx.sql`.

After this, he creates a pull request to alter schema with both `folder1/{{DB_NAME}}_00001_update.sql` and `folder2/{{DB_NAME}}_00001_another_update.sql` 
In this scenario, we may need to use different repositories to check different SQL files.

However, we can only use one repository to take the VCS API calls.

Reply:
Per offline chat, you're right. Closing.